### PR TITLE
enhancement: boolean field to support ignore_malformed

### PR DESCRIPTION
**Pull request by devnamrits**
This PR enables ignore_malformed to be accepted as an option in properties of mappings. 

```
{
    "mappings": {
        "properties": {
            "number_one": {
                "type": "integer"
            },
            "check": {
                "type": "boolean",
                "ignore_malformed": true
            }
        }
    }
}
```

Initially the `"ignore_malformed": true` was used to raise an exception. Now the index gets created and accepts non-boolean value as well.

It resolves the issue for: #89542

### CODE REVIEW

**Comment by devnamrits** [comment]
@javanna Please check. Also, I need to change the current tests as well and add test cases as well if required. Can you tell me how to proceed regarding tests?

**Review by javanna — COMMENTED** [review]
I left a couple of comments. but this looks quite close to me. I am going to start a test run to double check that I am not missing anything.

**Comment by javanna** [comment]
@elasticmachine test this please

**Comment by javanna** [comment]
@elasticmachine update branch

**Comment by javanna** [comment]
@elasticmachine retest this please

**Comment by javanna** [comment]
One more thing I forgot in my review above is the need to update docs. The boolean field needs to be added to the list of fields that support the option in https://github.com/elastic/elasticsearch/blob/main/docs/reference/mapping/params/ignore-malformed.asciidoc , and the boolean field type docs page needs updating too (https://github.com/elastic/elasticsearch/blob/main/docs/reference/mapping/types/boolean.asciidoc) .

File: server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java [hunk 1]
```diff
@@ -104,17 +105,23 @@ public static class Builder extends FieldMapper.Builder {
...
             this.scriptCompiler = Objects.requireNonNull(scriptCompiler);
             this.indexCreatedVersion = Objects.requireNonNull(indexCreatedVersion);
-            this.script.precludesParameters(nullValue);
+            this.ignoreMalformed = Parameter.explicitBoolParam(
+                "ignore_malformed",
+                true,
+                m -> toType(m).ignoreMalformed,
+                ignoreMalformedByDefault
+            );
+            this.script.precludesParameters(ignoreMalformed, nullValue);
```
[the remark this answers is not in the document]
Comment (devnamrits) [reply server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java:118]: Should I modify the following BooleanFieldMapperTests#testScriptAndPrecludedParameters test?

File: server/src/test/java/org/elasticsearch/index/mapper/BooleanScriptFieldTypeTests.java [hunk 2]
```diff
@@ -314,7 +314,7 @@ protected Query randomTermsQuery(MappedFieldType ft, SearchExecutionContext ctx)
     }
 
     public void testDualingQueries() throws IOException {
```
[the remark this answers is not in the document]
Comment (devnamrits) [reply server/src/test/java/org/elasticsearch/index/mapper/BooleanScriptFieldTypeTests.java:316]: Should I write the new tests within this PR itself? Or should I raise it another PR?

**Comment by devnamrits** [comment]
@javanna Thanks for reviewing my code. I've asked my doubts in the comments.

Should the documentation changes also be done within this PR? 

It would be great if you could list actionable items for this PR to be moved ahead for merging.

File: server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (javanna) [reply server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java:118]: yes please

File: server/src/test/java/org/elasticsearch/index/mapper/BooleanScriptFieldTypeTests.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (javanna) [reply server/src/test/java/org/elasticsearch/index/mapper/BooleanScriptFieldTypeTests.java:316]: as part of this PR ;)

**Comment by javanna** [comment]
hi @devnamrits , thank you for working on this.

> Should the documentation changes also be done within this PR?

Yes as part of this PR, please.

**Comment by devnamrits** [comment]
Okay then, making the necessary changes as discussed above.

Also, just wanted feedback, code committed for ignore_malformed till now is fine??

**Comment by javanna** [comment]
> Also, just wanted feedback, code committed for ignore_malformed till now is fine??

Yep, like I said above, it looks good to me.

**Comment by amyjtechwriter** [comment]
Hey @devnamrits do you need a hand updating the documentation pages?

**Comment by amyjtechwriter** [comment]
Hi @javanna I've pushed some changes to the docs. Could I ask you or a team member to review them and see if they are correct please?

**Review by amyjtechwriter — APPROVED** [review]
LGTM

**Comment by devnamrits** [comment]
Hi @amyjtechwriter, Thanks for merging my PR. I've been off for quite time due to some personal reasons. I'm thinking to pick up some new issues. Can you suggest some??

cc: @javanna

**Comment by DaveCTurner** [comment]
Unfortunately I have had to revert this change because it did not receive a full CI run and has some remaining issues. There's no way for me to reopen this PR, so @devnamrits could you create a new PR so we can continue working on this?

**Comment by javanna** [comment]
@amyjtechwriter the code part was not ready to be merged. I am going to revert this change.

**Comment by devnamrits** [comment]
> Unfortunately I have had to revert this change because it did not receive a full CI run and has some remaining issues. There's no way for me to reopen this PR, so @devnamrits could you create a new PR so we can continue working on this?

Creating a new one.

**Pull request merged**